src/common/dtpthread.c

```diff
@@ -0,0 +1,52 @@
+#include <sys/resource.h>
+
+#include "common/dtpthread.h"
+
+gboolean dt_openmp_init_stacksize(void)
+{
+  pthread_attr_t attr;
+  size_t stacksize;
+  gboolean ret=TRUE;
+
+  pthread_attr_init(&attr);
+  pthread_attr_getstacksize(&attr, &stacksize);
+  
+  const gchar *openmp_stacksize_env=openmp_stacksize_env=g_getenv("OMP_STACKSIZE");
+  
+  gchar *s=g_strdup_printf("%dB", SAFESTACKSIZE);
+  
+  if (stacksize < SAFESTACKSIZE && g_strcmp0(openmp_stacksize_env, s) != 0 )
```

Comment:
Given the info about this environment variable found in  https://gcc.gnu.org/onlinedocs/libgomp/OMP_005fSTACKSIZE.html and http://www.openmp.org/mp-documents/openmp-4.5.pdf (section 4.7 on page 298), it seems to me that whichever function retrieves this environment variable MUST process B/K/M/G suffixes as well as non-suffixed values.

Something along these lines:

char *pval = NULL;
uint64_t value = 0;

if ((pval = getenv("OMP_STACKSIZE")) == NULL) {
	value = MINIMUM_SAFESTACKSIZE;
} else {
	if (isascii(pval[strlen(pval)])) {
		/* modifier processing required */
		...
	} else {
		value = atol(pval);
	}
}
return (value);

and then use the returned value in setting pthread_attrs.